psi4/driver/driver_nbody.py

```diff
@@ -900,15 +899,66 @@ def get_results(self, results={}):
         nbody_results['intermediates2'] = {str(k): v['properties']["return_energy"] for k, v in results_list.items()}
 
         if ptype is not None:
-            nbody_results['intermediates_ptype'] = {'PTYPE ' + str(k): v for k, v in ptype.items()}
+            nbody_results['intermediates_ptype'] = {str(k): v for k, v in ptype.items()}
 
         return nbody_results
 
+    def get_results(self):
+        """Return nbody results as nbody-flavored QCSchema."""
+
+        results = self.prepare_results()
+        ret_energy, ret_ptype, ret_gradient = results.pop('ret_energy'), results.pop('ret_ptype'), results.pop('ret_gradient', None)
+
+        # load QCVariables
+        qcvars = {
+            'NUCLEAR REPULSION ENERGY': self.molecule.nuclear_repulsion_energy(),
+        }
+
+        for k, val in results.items():
+            for k2, val2 in val.items():
+                if isinstance(val2, np.ndarray):
+                    val[k2] = val2.tolist()
+            qcvars[k] = val
+
+        qcvars['CURRENT ENERGY'] = ret_energy
+        if self.driver == 'gradient':
+            qcvars['CURRENT GRADIENT'] = ret_ptype.tolist()
+        elif self.driver == 'hessian':
+            qcvars['CURRENT GRADIENT'] = ret_gradient.tolist()
+            qcvars['CURRENT HESSIAN'] = ret_ptype.tolist()
+
+        component_results = self.dict()['task_list']
+        for k1, val1 in component_results.items():
+            for k2 in val1:
+                val1[k2]['molecule'] = val1[k2]['molecule'].to_schema(dtype=1)['molecule']
+
+        data = {
+            'driver': self.driver,
+            'molecule': self.molecule.to_schema(dtype=1)['molecule'],
+            'properties': {
+                'calcinfo_natom': self.molecule.natom(),
+                'nuclear_repulsion_energy': self.molecule.nuclear_repulsion_energy(),
+                'return_energy': ret_energy,
+            },
+            'provenance': p4util.provenance_stamp(__name__),
+            'psi4:qcvars': qcvars,
+            #'raw_output': None,
+            'return_result': ret_energy if self.driver == 'energy' else ret_ptype.tolist(),
```

Comment:
@alenaizan All this casting between list, ndarray, and Matrix can get excessive. As long as a thing doesn't need to get written to json, I'm defaulting (for cbs) to leaving it as ndarray (non-flat). Then there's a `unnp` fn (in task_base.py) for now to take a nested dictionary and convert any ndarray in it to flat lists just before declaring the dict ready for json. And there's a `plump` fn (still in driver_cbs.py) that can re-inflate lists to ndarray with sensible shape.

Reply:
I fixed this and moved ```plump``` to task_base.py